stdlib/public/core/StringBridge.swift

```diff
@@ -423,25 +423,37 @@ private var expectedConstantTagValue:UInt {
 }
 #endif
 
+private enum ConstantTaggedStringValidationResult {
+  case notApplicable
+  case valid
+  case invalid
+}
+
 @inline(__always)
-private func formConstantTaggedCocoaString(
-  untaggedCocoa: _CocoaString
-) -> AnyObject? {
-#if !arch(arm64)
-  return nil
-#else
+private func validateConstantTaggedCocoa(_ cocoaString: _CocoaString) ->
+  ConstantTaggedStringValidationResult {
+  #if !arch(arm64)
+  return .notApplicable
+  #else
+  guard _isObjCTaggedPointer(cocoaString) else {
+    return .notApplicable
+  }
 
-  let constantPtr:UnsafeRawPointer = Builtin.reinterpretCast(untaggedCocoa)
+  let taggedValue = unsafeBitCast(cocoaString, to: UInt.self)
```

Comment:
I don't know if `UInt(bitPattern: ObjectIdentifier(x))` would be considered better. @lorentey ?